src/thunderbots/software/grsim_communication/motion_controller.cpp

```diff
@@ -32,6 +34,11 @@ MotionController::Velocity MotionController::bangBangVelocityController(
         throw std::invalid_argument(
             "GrSim Motion controller received a negative delta time");
     }
+    else if (velocity_request)
+    {
+        robot_velocities.linear_velocity  = requested_linear_velocty;
```

Comment:
Extra logic should be added here to smoothly ramp the velocities up and down, otherwise we will immediately apply a velocity in grsim which will give wonky results (because _instantaneous acceleration_ isn't great or realistic).

It can be much simpler than everything below. I think something as simple as adding/subtracting acceleration to the current velocity and clamping it would be sufficient.

Reply:
When you say adding acceleration, do you mean the maximum possible acceleration so the robot would get to the desired velocity as quick as realistically possible?

Reply:
Yup :+1: Something like adding `MAX_ACCELERATION * time_step` velocity each time to reach the requested velocity as quickly as possible.